src/main/java/chess/chessgame/Chessboard.java

```diff
@@ -0,0 +1,138 @@
+
+        validateBlank(source);
+        validateTurn(source, turn);
+        validateColor(source, target);
+        validateMovable(movingPosition, source, target);
+        validateMiddlePosition(movingPosition, source);
+    }
+
+    private void validateBlank(Piece piece) {
+        if (piece.isSameType(Type.BLANK)) {
+            throw new IllegalArgumentException("이동하려는 위치에 기물이 없습니다.");
+        }
+    }
+
+    private void validateTurn(Piece source, Turn turn) {
+        if (!turn.isRightTurn(source.getColor())) {
+            throw new IllegalArgumentException("상대편의 기물은 움직일 수 없습니다.");
+        }
+    }
+
+    private void validateColor(Piece source, Piece target) {
+        if (source.isSameColor(target.getColor())) {
+            throw new IllegalArgumentException("상대편의 기물으로만 이동할 수 있습니다.");
+        }
+    }
+
+    private void validateMovable(MovingPosition movingPosition, Piece source, Piece target) {
+        if (!target.isSameType(Type.BLANK) && isDiagonalPawn(movingPosition, source)) {
+            return;
+        }
+        if (source.isMovable(movingPosition)) {
+            return;
+        }
+        throw new IllegalArgumentException("움직일 수 없는 기물입니다.");
+    }
+
+    private boolean isDiagonalPawn(MovingPosition movingPosition, Piece source) {
+        if (!source.isSameType(Type.PAWN)) {
+            return false;
+        }
+        Pawn pawn = (Pawn) source;
+        return pawn.isDiagonal(movingPosition);
+    }
+
+    private void validateMiddlePosition(MovingPosition movingPosition, Piece source) {
+        source.computeMiddlePosition(movingPosition).forEach(
+                middlePosition -> validateMiddleBlank(middlePosition)
+        );
+    }
+
+    private void validateMiddleBlank(Position position) {
+        if (!findPiece(position.getX(), position.getY()).isSameType(Type.BLANK)) {
+            throw new IllegalArgumentException("가로막는 기물이 있습니다.");
+        }
+    }
+
+    private void movePiece(MovingPosition movingPosition) {
+        Piece source = findPiece(movingPosition.getFromX(), movingPosition.getFromY());
+
+        board.get(movingPosition.getToX()).set(movingPosition.getToY(), source);
+        board.get(movingPosition.getFromX()).set(movingPosition.getFromY(), new Blank());
+    }
+
+    public double computeScore(Color color, double minusScoreOfSameColumnPawn) {
+        double score = computeTotalScore(color);
+
+        for (int i = 0; i < board.size(); i++) {
+            int duplicatedPawn = computePawnCount(i, color);
+            score -= computeMinusScore(duplicatedPawn, minusScoreOfSameColumnPawn);
+        }
+
+        return score;
+    }
+
+    private double computeTotalScore(Color color) {
+        double score = 0;
+        for (List<Piece> list : board) {
+            score += list.stream()
+                    .filter(piece -> piece.isSameColor(color))
+                    .mapToDouble(Piece::getScore)
+                    .sum();
+        }
+        return score;
```

Comment:
```suggestion
return board.stream()
                .flatMap(Collection::stream)
                .filter(piece -> piece.isSameColor(color))
                .mapToDouble(Piece::getScore)
                .sum();
```
현재구조에서는 개선은 위처럼하면 될것같은데, 인스턴스 변수가 달라진다면 그에 맞게 변경하셔야될거에요~

Reply:
자료형이 map으로 변하게 되면서 쓰지않게 되었지만 flatMap에 대해 공부했습니다! 2차원 배열 혹은 리스트를 1차원 배열의 stream으로 낮춰주다니 정말 유용하네요. 제이 덕분에 새롭게 알게되었어요. 감사합니다😃